mantis-publish/mantis-publish-core/src/main/java/io/mantisrx/publish/AbstractSubscriptionTracker.java

```diff
@@ -59,148 +67,167 @@ public AbstractSubscriptionTracker(MrePublishConfiguration mrePublishConfigurati
+
+
+	/**
+	 * Determines which job clusters (source jobs) are currently mapped to the
+	 * specified application.
+	 *
+	 * @param streamJobClusterMap A {@link Map} of stream name to job cluster.
+	 * @param registeredStreams A {@link Set} of registered streams.
+	 *
+	 * @return A {@link Set} of job clusters relevant to this application.
+	 * */
+	private Set<String> getRelevantJobClusters(Map<String, String> streamJobClusterMap, Set<String> registeredStreams) {
+		Set<String> jobClustersToFetch = new HashSet<>();
+
+		for (Map.Entry<String, String> e : streamJobClusterMap.entrySet()) {
+			String streamName = e.getKey();
+			LOG.debug("processing stream {} and currently registered Streams {}", streamName, registeredStreams);
+			if (registeredStreams.contains(streamName)
+					|| StreamJobClusterMap.DEFAULT_STREAM_KEY.equals(streamName)) {
+				jobClustersToFetch.add(e.getValue());
+			} else {
+				LOG.warn("No server side mappings found for one or more streams {} ", registeredStreams);
+				LOG.debug("will not fetch subscriptions for un-registered stream {}", streamName);
+			}
+		}
+
+		return jobClustersToFetch;
+	}
+
+	@Override
+	public void refreshSubscriptions() {
+		refreshSubscriptionInvokedCount.increment();
+
+		boolean mantisPublishEnabled = mrePublishConfiguration.isMREClientEnabled();
+		final Set<String> registeredStreams = streamManager.getRegisteredStreams();
+
+		if (mantisPublishEnabled && !registeredStreams.isEmpty()) {
+			final Map<String, String> streamJobClusterMap =
+				jobDiscovery.getStreamNameToJobClusterMapping(mrePublishConfiguration.appName());
+			Set<String> jobClustersToFetch = getRelevantJobClusters(streamJobClusterMap, registeredStreams);
+
+			Set<MantisServerSubscription> allSubscriptions = new HashSet<>();
+			for (String jobCluster : jobClustersToFetch) {
+				try {
+					Optional<MantisServerSubscriptionEnvelope> subsEnvelopeO = fetchSubscriptions(jobCluster);
+					if (subsEnvelopeO.isPresent()) {
+						MantisServerSubscriptionEnvelope subsEnvelope = subsEnvelopeO.get();
+						allSubscriptions.addAll(subsEnvelope.getSubscriptions());
+						refreshSubscriptionSuccessCount.increment();
+					} else {
+						refreshSubscriptionFailedCount.increment();
+					}
+				} catch (Exception ex) {
+					LOG.info("refresh subscriptions failed for {}", jobCluster, ex);
+					refreshSubscriptionFailedCount.increment();
+				}
+			}
+
+			propagateSubscriptionChanges(allSubscriptions);
+		} else {
+			LOG.debug("subscription refresh skipped (client enabled {} registered streams {})",
+					mantisPublishEnabled, registeredStreams);
+		}
+	}
+
+	protected Set<Subscription> getCurrentSubscriptions() {
+
+		return streamManager
+			.getRegisteredStreams()
+			.stream()
+			.flatMap(streamName -> streamManager.getStreamSubscriptions(streamName).stream())
+			.collect(Collectors.toSet());
+	}
+
+	// TODO: This is only present for unit tests, should be removed.
```

Comment:
We should just remove this now, and fix any tests that are using it. Those are tests are likely not testing the right assumptions anymore.